src/macroexpand.scm

```diff
@@ -340,8 +340,22 @@
 
 (define (reescape ux x)
   (if (and (pair? x) (eq? (car x) 'escape))
-    (reescape '(escape ,ux) (cadr x)))
-    ux)
+    (reescape `(escape ,ux) (cadr x))
+    ux))
```

Comment:
I think `reescape` always returned `ux`, this is no longer the case with this fix.